packages/core/src/m3/factory.ts

```diff
@@ -32,44 +38,60 @@ export class LanguageFactory {
     }
 
 
-    // TODO  also take care of containment immediately? (then most calls to .having<X> would be unnecessary/wrong)
-
     annotation(name: string, extends_?: SingleRef<Annotation>) {
```

Comment:
question due to my ignorance with TS: shouldn't we specify the return type for this and the following methods?

Reply:
Actually: we should. Didn't do that so far because the API was involving a bit, still.

Reply:
Oh, and specifying a return type is optional in TS as TS derives it from the statement block, but one can certainly consider it to be good manners to specify it nevertheless.